httpcore/_backends/asyncio.py

```diff
@@ -195,8 +195,10 @@ async def aclose(self) -> None:
 
     def is_readable(self) -> bool:
         transport = self.stream_reader._transport  # type: ignore
-        sock: socket.socket = transport.get_extra_info("socket")
-        return is_socket_readable(sock.fileno())
+        sock: Optional[socket.socket] = transport.get_extra_info("socket")
+        # If socket was detatched from the transport, most likely connection was reset.
+        # Hence make it readable to notify users to poll the socket.
+        return not sock or is_socket_readable(sock.fileno())
```

Comment:
(Nit) Using explicit `None` checks:

```suggestion
        return sock is None or is_socket_readable(sock.fileno())
```